### PR TITLE
Add reuse and related files

### PR DESCRIPTION
**Pull request by csyonghe**

### CODE REVIEW

**Review by aleino-nv — APPROVED** [review]

File: .reuse/dep5 [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+Files: tests/fcpw/*
+Copyright: 2020 Rohan Sawhney
+License: MIT
+
+Files: external/renderdoc_app.h
+Copyright: 2019-2021 Baldur Karlsson
+License: MIT
+
+Files: external/dxc/dxcapi.h
```
Comment (expipiplus1) [inline .reuse/dep5:33]: same for this, the header says UoI open source license

File: .reuse/dep5 [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+Copyright: Microsoft Corporation. All rights reserved.
+License: MIT
+
+Files: source/core/slang-crypto.cpp
```
Comment (expipiplus1) [inline .reuse/dep5:37]: should this include the corresponding header too?

**Review by expipiplus1 — CHANGES_REQUESTED** [review]
Have we verified that we have the consent of all contributors to relicense their changes?

File: .reuse/dep5 [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+Copyright: 2016, Carnegie Mellon University and NVIDIA Corporation
+           2024 The Khronos Group, Inc.
+License: Apache-2.0 WITH LLVM-exception
+
+Files: docs/_includes/anchor_headings.html
+Copyright: 2018 Vladimir "allejo" Jimenez
+License: Apache-2.0 WITH LLVM-exception
+
+Files: external/spirv/spirv.h
+       external/glext.h
```
[the remark this answers is not in the document]
Comment (csyonghe) [reply .reuse/dep5:16]: No, these needs to be MIT.

File: .reuse/dep5 (hunk 1 above)
Comment (csyonghe) [reply .reuse/dep5:33]: needs to change.

File: .reuse/dep5 (hunk 2 above)
Comment (csyonghe) [reply .reuse/dep5:37]: We did not incorporate any original header file, so this should be fine.

File: .reuse/dep5 (hunk 1 above)
Comment (csyonghe) [reply .reuse/dep5:33]: Done.

File: .reuse/dep5 (hunk 3 above)
[the remark this answers is not in the document]
Comment (csyonghe) [reply .reuse/dep5:16]: This should be fine, mark as resolved.

**Review by expipiplus1 — APPROVED** [review]

**Pull request merged**